source/solvers/vof.cc

```diff
@@ -221,6 +470,105 @@ VolumeOfFluid<dim>::assemble_system_rhs()
 }
 
+template <int dim>
+void
+VolumeOfFluid<dim>::assemble_system_rhs_dg()
+{
+  const DoFHandler<dim> *dof_handler_fd =
+    multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
+
+  auto scratch_data =
+    VOFScratchData<dim>(this->simulation_control,
+                        this->simulation_parameters.physical_properties_manager,
+                        *this->fe,
+                        *this->cell_quadrature,
+                        *this->face_quadrature,
+                        *this->mapping,
+                        dof_handler_fd->get_fe());
+
+  StabilizedDGMethodsCopyData copy_data(this->fe->n_dofs_per_cell(),
+                                        this->cell_quadrature->size());
+
+  const auto cell_worker =
+    [&](const typename DoFHandler<dim>::active_cell_iterator &cell,
+        VOFScratchData<dim>                                  &scratch_data,
+        StabilizedDGMethodsCopyData                          &copy_data) {
+      this->assemble_local_system_rhs(cell, scratch_data, copy_data);
+    };
+
+  const auto boundary_worker =
+    [&]([[maybe_unused]] const typename DoFHandler<dim>::active_cell_iterator
+                                                     &cell,
+        [[maybe_unused]] const unsigned int          &face_no,
+        [[maybe_unused]] VOFScratchData<dim>         &scratch_data,
+        [[maybe_unused]] StabilizedDGMethodsCopyData &copy_data) {};
+
+  const auto face_worker =
+    [&](const typename DoFHandler<dim>::active_cell_iterator &cell,
+        const unsigned int                                   &face_no,
+        const unsigned int                                   &sub_face_no,
+        const typename DoFHandler<dim>::active_cell_iterator &neigh_cell,
+        const unsigned int                                   &neigh_face_no,
+        const unsigned int                                   &neigh_sub_face_no,
+        VOFScratchData<dim>                                  &scratch_data,
+        StabilizedDGMethodsCopyData                          &copy_data)
+
+  {
+    scratch_data.reinit_internal_face(cell,
+                                      face_no,
+                                      sub_face_no,
+                                      neigh_cell,
+                                      neigh_face_no,
+                                      neigh_sub_face_no,
+                                      this->evaluation_point);
+
+    copy_data.face_data.emplace_back();
+    auto &copy_data_face = copy_data.face_data.back();
+    copy_data_face.joint_dof_indices =
+      scratch_data.fe_interface_values_vof.get_interface_dof_indices();
+    copy_data_face.face_rhs.reinit(scratch_data.n_interface_dofs);
+
+    // Gather velocity information at the face to properly advect
+    // First gather the dof handler for the fluid dynamics
+    const DoFHandler<dim> *dof_handler_fluid =
+      multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
```

Comment:
Repeats line 477
```suggestion
```

Reply:
Maybe it is needed because it is in a lamba function

Reply:
yeah it's needed in the lambda